src/hotspot/share/opto/parse.hpp

```diff
@@ -603,4 +603,55 @@ class Parse : public GraphKit {
 #endif
 };
 
+// Specialized uncommon_trap of unstable_if, we have 2 optimizations for them:
+//   1. suppress trivial Unstable_If traps
+//   2. use next_bci of _path to update live locals.
+class UnstableIfTrap {
```

Comment:
What about moving this information into `CallStaticJavaNode`?

Reply:
I think CallStaticJavaNode is popular.  uncommon_trap/unstable_if is just a special case. that's why factor out and use a dedicated class for it.